src/net.rs

```diff
@@ -278,98 +330,180 @@ struct Actor {
-                    inc!(Metrics, actor_tick_endpoint);
-                    current_addresses = new_addresses;
-                    let peer_data = our_peer_data(&self.endpoint, &current_addresses)?;
-                    self.handle_in_event(InEvent::UpdatePeerData(peer_data), Instant::now()).await?;
                 }
-                Some(_relay_url) = home_relay_stream.next() => {
-                    let peer_data = our_peer_data(&self.endpoint, &current_addresses)?;
-                    self.handle_in_event(InEvent::UpdatePeerData(peer_data), Instant::now()).await?;
-                }
-                (peer_id, res) = self.dialer.next_conn() => {
-                    trace!(?i, "tick: dialer");
-                    inc!(Metrics, actor_tick_dialer);
-                    match res {
-                        Ok(conn) => {
-                            debug!(peer = ?peer_id, "dial successful");
-                            inc!(Metrics, actor_tick_dialer_success);
-                            self.handle_connection(peer_id, ConnOrigin::Dial, conn);
-                        }
-                        Err(err) => {
-                            warn!(peer = ?peer_id, "dial failed: {err}");
-                            inc!(Metrics, actor_tick_dialer_failure);
-                        }
+            },
+            Some((key, (topic, command))) = self.command_rx.next(), if !self.command_rx.is_empty() => {
+                trace!(?i, "tick: command_rx");
+                self.handle_command(topic, key, command).await?;
+            },
+            Some(new_addresses) = direct_addresses_stream.next() => {
+                trace!(?i, "tick: new_endpoints");
+                inc!(Metrics, actor_tick_endpoint);
+                *current_addresses = new_addresses;
+                self.handle_addr_update(current_addresses).await?;
+            }
+            Some(_relay_url) = home_relay_stream.next() => {
+                trace!(?i, "tick: new_home_relay");
+                self.handle_addr_update(current_addresses).await?;
+            }
+            (peer_id, res) = self.dialer.next_conn() => {
+                trace!(?i, "tick: dialer");
+                inc!(Metrics, actor_tick_dialer);
+                match res {
+                    Ok(conn) => {
+                        debug!(peer = ?peer_id, "dial successful");
+                        inc!(Metrics, actor_tick_dialer_success);
+                        self.handle_connection(peer_id, ConnOrigin::Dial, conn);
                     }
-                }
-                event = self.in_event_rx.recv() => {
-                    trace!(?i, "tick: in_event_rx");
-                    inc!(Metrics, actor_tick_in_event_rx);
-                    match event {
-                        Some(event) => {
-                            self.handle_in_event(event, Instant::now()).await.context("in_event_rx.recv -> handle_in_event")?;
-                        }
-                        None => unreachable!()
```

Comment:
@rklaehn this is the original line, unchanged. Since the code moved in a way such that it's now with a a different indentation it's a bit hard to parse. To specifically answer your question I don't really know

Reply:
yeah, I was just curious. I saw that this wasn't your code that introduced this...